src/states/joining_peer.rs

```diff
@@ -168,6 +206,141 @@ impl JoiningPeer {
+    }
+
+    fn send_bootstrap_request(&mut self, dst: SocketAddr) {
+        let stage = match &mut self.stage {
+            Stage::Bootstrapping(stage) => stage,
+            Stage::Joining(_) => unreachable!(),
+        };
+
+        if !stage.pending_requests.insert(dst) {
+            return;
+        }
+
+        let token = self.timer.schedule(BOOTSTRAP_TIMEOUT);
+        let _ = stage.timeout_tokens.insert(token, dst);
+
+        let destination = match &stage.relocate_details {
+            Some(details) => *details.destination(),
+            None => *self.name(),
+        };
+
+        debug!("{} Sending BootstrapRequest to {}.", self, dst);
+        self.send_direct_message(&dst, Variant::BootstrapRequest(destination));
+    }
+
+    fn reconnect_to_new_section(&mut self, new_conn_infos: Vec<SocketAddr>) {
+        match &mut self.stage {
+            Stage::Bootstrapping(stage) => {
+                for addr in stage.pending_requests.drain() {
+                    self.network_service.disconnect(addr);
+                }
+
+                stage.timeout_tokens.clear();
+
+                for conn_info in new_conn_infos {
+                    self.send_bootstrap_request(conn_info);
+                }
+            }
+            Stage::Joining(_) => unreachable!(),
+        }
+    }
+
+    fn send_join_requests(&mut self) {
+        let stage = match &mut self.stage {
+            Stage::Bootstrapping(_) => unreachable!(),
+            Stage::Joining(stage) => stage,
+        };
+
+        let relocate_payload = match &stage.join_type {
+            JoinType::First { .. } => None,
+            JoinType::Relocate(payload) => Some(payload),
+        };
+
+        let elders_version = stage.elders_info.version();
+        let messages: Vec<_> = stage
+            .elders_info
+            .member_nodes()
+            .map(|dst| {
+                let join_request = JoinRequest {
+                    elders_version,
+                    relocate_payload: relocate_payload.cloned(),
+                };
+
+                (dst.clone(), Variant::JoinRequest(Box::new(join_request)))
+            })
+            .collect();
+
+        for (dst, variant) in messages {
+            info!("{} - Sending JoinRequest to {}", self, dst);
+            self.send_direct_message(dst.peer_addr(), variant);
+        }
+    }
+
+    fn join_section(&mut self, elders_info: EldersInfo) -> Result<()> {
+        let stage = match &mut self.stage {
+            Stage::Bootstrapping(stage) => stage,
+            Stage::Joining(_) => unreachable!(),
+        };
+
+        let relocate_details = stage.relocate_details.take();
+        let destination = match &relocate_details {
+            Some(details) => *details.destination(),
+            None => *self.name(),
+        };
+        let old_full_id = self.full_id.clone();
+
+        // Use a name that will match the destination even after multiple splits
+        let extra_split_count = 3;
```

Comment:
Well it's sort of already is, it just uses `let` instead of `const`. But again, this is one of those things that I just moved over from the previous code and did not change. For the sake of keeping the amount of change in the PR minimal, I'd prefer to not change stuff that's not necessary for the purpose of this PR, no matter how trivial that change might be.